File: EVSDKDemo/Streamer/EVConfigBeautyLevelViewController.m
//
//  EVConfigBeatyLevelViewController.m
//  EVSDKDemo
//
//  Created by Lcrnice on 2017/6/21.
//  Copyright © 2017年 cloudfocous. All rights reserved.
//

#import "EVConfigBeautyLevelViewController.h"

@interface EVConfigBeautyLevelViewController ()
@property (weak, nonatomic) IBOutlet UISegmentedControl *seg;
@property (weak, nonatomic) IBOutlet UISwitch *switchBtn;

@end

@implementation EVConfigBeautyLevelViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.seg.selectedSegmentIndex = self.index;
    self.switchBtn.on =
    self.switchBtn.selected =
    self.isBeautyEnable;
    
    self.seg.enabled = self.isBeautyEnable;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


+ (instancetype)instanceVC {
    UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    return [storyBoard instantiateViewControllerWithIdentifier:@"EVConfigBeautyLevelViewController"];
}
- (IBAction)onBackgroundTouchDown:(id)sender {
    [self dismiss];
}

- (void)showInViewController:(UIViewController *)parentViewController {
    if (self.parentViewController == parentViewController) {
        return;
    }
    
    [parentViewController addChildViewController:self];
    self.view.alpha = 0.0;
    self.view.frame = parentViewController.view.bounds;
    [parentViewController.view addSubview:self.view];
    
    [self didMoveToParentViewController:parentViewController];
    
    [UIView animateWithDuration:0.25 animations:^{
        self.view.alpha = 1.0;
    }];
}

- (void)dismiss {
    [UIView animateWithDuration:0.25 animations:^{
        self.view.alpha = 0.0;
    } completion:^(BOOL finished) {
        [self willMoveToParentViewController:nil];
        [self.view removeFromSuperview];
        [self removeFromParentViewController];
    }];
}


- (IBAction)segChanged:(UISegmentedControl *)sender {
    NSInteger index = sender.selectedSegmentIndex;
    !self.segValueChanged ? : self.segValueChanged(index+1);
}
- (IBAction)switchChanged:(UISwitch *)sender {
    sender.selected = !sender.selected;
    BOOL selected = sender.selected;
    self.seg.enabled = selected;
    !self.switchValueChanged ? : self.switchValueChanged(selected);
    if (selected) {
        self.seg.selectedSegmentIndex = 2;
        !self.segValueChanged ? : self.segValueChanged(3);
    }
}

@end

File: EVSDKDemo_Swift/Streamer/EVConfigBeautyLevelViewController.swift
//
//  EVConfigBeautyLevelViewController.swift
//  EVSDKDemo
//
//  Created by Lcrnice on 2017/8/10.
//  Copyright © 2017年 cloudfocous. All rights reserved.
//

import UIKit

class EVConfigBeautyLevelViewController: UIViewController {
  
  var index: Int = 2
  var isBeautyEnable = false
		
  var segValueChanged: ((Int) -> Void)?
  var switchValueChanged: ((Bool) -> Void)?
  
  @IBOutlet weak var switchBtn: UISwitch!
  @IBOutlet weak var seg: UISegmentedControl!
  
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    seg.selectedSegmentIndex = index
    switchBtn.isSelected = isBeautyEnable
    switchBtn.isOn = isBeautyEnable
    seg.isEnabled = isBeautyEnable
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
  }
  
  class func instanceVC() -> EVConfigBeautyLevelViewController {
    let storyboard = UIStoryboard(name: "Main", bundle: nil)
    return storyboard.instantiateViewController(withIdentifier: "EVConfigBeautyLevelViewController") as! EVConfigBeautyLevelViewController
  }
  
  func showInViewController(parentViewController vc: UIViewController) {
    if self.parent == vc {
      return
    }
    
    vc.addChildViewController(self)
    self.view.alpha = 0
    self.view.frame = vc.view.frame
    vc.view.addSubview(self.view)
    
    self.didMove(toParentViewController: vc)
    
    UIView.animate(withDuration: 0.25, animations: {
      self.view.alpha = 1
    })
  }
  
  private func dismissVC() {
    UIView.animate(withDuration: 0.25, animations: {
      self.view.alpha = 0
    }, completion: { finished in
      self.willMove(toParentViewController: nil)
      self.view .removeFromSuperview()
      self .removeFromParentViewController()
    })
  }
  
  @IBAction func onBackgroundTouchDown(_ sender: Any) {
    self.dismissVC()
  }
  
  @IBAction func switchChanged(_ sender: UISwitch) {
    sender.isSelected = !sender.isSelected
    seg.isEnabled = sender.isSelected
    
    if switchValueChanged != nil {
      switchValueChanged!(sender.isSelected)
    }
    
    if sender.isSelected {
      seg.selectedSegmentIndex = 2
      if segValueChanged != nil {
        segValueChanged!(3)
      }
    }
  }
  
  @IBAction func segChanged(_ sender: UISegmentedControl) {
    let index = sender.selectedSegmentIndex
    if segValueChanged != nil {
      segValueChanged!(index)
    }
  }
  /*
   // MARK: - Navigation
   
   // In a storyboard-based application, you will often want to do a little preparation before navigation
   override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
   // Get the new view controller using segue.destinationViewController.
   // Pass the selected object to the new view controller.
   }
   */
  
}
